### DISCUSSION TITLE
SSH connection bugs

### DISCUSSION THREAD

**Discussion by AASabancev** [electerm/electerm#3602] [General]
Hello!

1. connection to SSH via hops (via connection to another server) does not work. For example, you connect to the first server via an external IP address, and to the second server - via an internal IP between them). Also, no log is visible - at what stage the connection broke, when connecting to the first server or the second

2 the password to the certificate does not work when connecting via SSH, the password is saved but still asks for it every time you connect

**Comment by zxdong262** [comment]
Created a issue to track this: #3603

**Comment by zxdong262** [comment]
As I tested, connection hopping do works, can you confirm all passwords you entered is right?

**Reply by AASabancev** [reply]
1 I created two connections in electerm
2 first - is a VM server address like 178.11.22.33 port 500, connection via password
3 check direct connection to SSH - working
4 second - is VMachine on first server, does not have global IP address, only local like 10.11.22.33, port 22 via password

5 checking connection from server to VM from 178.11.22.33 to 10.11.22.33 like command "ssh root@10.11.22.33", copied ip address from bookmarks, copied password, connected succesfuly

6 checking settings of second connection via hops
6.1 ip, user, password is right (checked before)
6.2 connection hopping tried in two ways
- by selecting from bookmarks
- by entering address/port/user/password directly
the connection is added to the table, but does not work

7 can you make sure that you do not have a direct connection to the second server? maybe you are connecting to it directly and not through an intermediate server?

after conducting additional tests I was convinced
1 there are two servers accessible to me directly,
2 if you select a connection through the other in one of them - technically, it creates the appearance of a successful connection
3 BUT, the connection is not made to the final server, but only to the intermediate one, and I see the console of the intermediate server.

**Reply by zxdong262** [reply]
Thank you for the detail explaination, I would check it

**Reply by zxdong262** [reply]
As I tested, use a internal network ip to do hopping would still works, the internal ip can not directly be accessed

**Reply by AASabancev** [reply]
Well, you don't need direct access to the internal address.
After all, that's why I connect to the second server through the first one.
And the first server already has access to the second one via the internal address.

In theory, first a SSH connection to the first server should be made, then a SSH connection to the second one. That's what hopping is all about
